### DISCUSSION TITLE
Redirect status code causing no redirect to occur

### DISCUSSION THREAD

**Discussion by jonnow** [fastify/fastify#4478] [Q&A] [answered]
Hey, I was looking at the [fastify-session authentication](https://github.com/fastify/example/tree/master/fastify-session-authentication) example when I noticed this. 
I have a post request to `/login` that does a redirect to the homepage. Without setting a status code this works without any issue. However, I wanted to send a status code along to then change the UI if it's a specific code (say 213 for arguments sake).

However, when I try to include a status code with the reply things seem to break as the redirect no longer happens.

Can anyone let me know if this is a known issue or perhaps I'm not doing something right with `fastify.redirect(statusCode, url)`?

```
// Homepage GET
  fastify.get('/', (request, reply) => {
    reply.type('text/html')
    reply.send(defaultPage(request.session.authenticated))
  });

// Login POST
fastify.post('/login', (request, reply) => {
    const { email, password } = request.body
    if (password === 'abcdef') {
      request.session.authenticated = true
      reply.redirect(213, '/')
      //reply.redirect('/') // <- This way works and redirects to the '/' GET but without the custom status code
    } else {
      reply.redirect(401, '/login')
    }
  });
```

**Comment by metcoder95** [comment, marked as answer]
Hey! Can we know which `HTTP Status` code you were using?
It is important to respect the `HTTP 3xx` redirect semantics as that will indicate to the browser how to handle the redirect and what to do next. If you're trying to execute a `redirect` by replying with a, e.g. `213`, as it falls into the range >=`200` < `300` for instance the browser will understand it is ok and nothing else has to be done, as the server indicated so.

For a redirect for being successful, it should fall in the range of > `300` depending on your feature need.

**Reply by metcoder95** [reply]
By default, `fastify` uses `302` as redirection status code

**Reply by jonnow** [reply]
I was trying 213 which didn't redirect. I've just tried it with 302 and that does redirect to the '/' route. However, the request.statusCode is still undefined rather than being 302. Should I be able to see this status code in the incoming request to '/'

**Reply by metcoder95** [reply]
By HTTP specs, the status code is not part of the HTTP Request semantics. As the `Status Code` is part of the HTTP Response and will remain with the browser, and then the browser executes the redirection constructing a new HTTP Request.

If you want to share data between redirects, the most common option is to share it by the query string of the redirected path (e.g. `reply.redirect('/redirect?myprop=myvalue')`)

**Reply by jonnow** [reply]
Great, thank you!
